test/integration-test-main.sh

```diff
@@ -2243,59 +2244,76 @@ function test_not_existed_dir_obj() {
     echo data2 | aws_cli s3 cp - "s3://${TEST_BUCKET_1}/${OBJECT_NAME_2}"
 
     # Top directory
+    #
+    # [NOTE]
+    # Immediately after uploading with the aws command, there is a case where
+    # the directory path cannot be found (404 error).
+    # In this case, it may be listed correctly after a few retries.
+    # This error is a response problem on the S3 server side, not a problem
+    # with the s3fs process, so we try again.
```

Comment:
I don't understand this comment -- does S3Proxy have a bug where directories are not immediately exposed?  If so I would like to understand this more so I can address that issue.

Reply:
The cause of this is still unknown, but I determined that it was a response problem to the API request, and tried to solve it with this loop.

I collected the logs when this test fails (it does fail occasionally) and the logs recorded a 404 error.
And I've tried a number of workarounds, but if I end up receiving a 404 error, I've seen it become 200 if I repeat the request multiple times.
A successful re-request after this interval may be #2319. (I may have made a mistake in the analysis of the 404 error above)

So we need to check again once #2319 is merged.

Reply:
This problem seems to persist regardless of the stat `--cached=never` option.

So I ran CI again with `alpine:3.17`, only running the `test_not_existed_dir_obj` test, and with `curldbg` option.
As a result of this execution, I got an error occurs in the `(*)` part below.
```

    #
    # Create files under not existed directory by aws command
    #
    local OBJECT_NAME_1; OBJECT_NAME_1="${DIR_NAME}/not_existed_dir_single/${TEST_TEXT_FILE}"
    local OBJECT_NAME_2; OBJECT_NAME_2="${DIR_NAME}/not_existed_dir_parent/not_existed_dir_child/${TEST_TEXT_FILE}"
    echo data1 | aws_cli s3 cp - "s3://${TEST_BUCKET_1}/${OBJECT_NAME_1}"
    echo data2 | aws_cli s3 cp - "s3://${TEST_BUCKET_1}/${OBJECT_NAME_2}"

    # Top directory
    # shellcheck disable=SC2010
    if ! ls -1 | grep -q '^not_existed_dir_single$'; then
        echo "Expect to find \"not_existed_dir_single\" directory, but it is not found"
        return 1;
    fi
    # shellcheck disable=SC2010
(*) if ! ls -1 | grep -q '^not_existed_dir_parent$'; then
        echo "Expect to find \"not_existed_dir_parent\" directory, but it is not found"
        return 1;
    fi
```
You can see the execution log at this time:
https://github.com/ggtakec/s3fs-fuse/actions/runs/6290707707